app/src/test/java/org/simple/clinic/instantsearch/InstantSearchUpdateTest.kt

```diff
@@ -315,14 +316,53 @@ class InstantSearchUpdateTest {
     val indiaNHIDInfoPayload = TestData.indiaNHIDInfoPayload(
         healthIdNumber = indiaNationalHealthID
     )
+
+    val patientPrefillInfo = indiaNHIDInfoPayload.toPatientPrefillInfo()
+
+    val identifier = Identifier(indiaNationalHealthID, IndiaNationalHealthId)
+
+    val ongoingNewPatientEntry = OngoingNewPatientEntry(
+        personalDetails = OngoingNewPatientEntry.PersonalDetails(
+            fullName = patientPrefillInfo.fullName,
+            dateOfBirth = patientPrefillInfo.dateOfBirth.format(DateTimeFormatter.ofPattern("dd/MM/yyyy")),
+            gender = Gender.Unknown("M"),
+            age = null),
+        address = OngoingNewPatientEntry.Address.BLANK.withColonyOrVillage(patientPrefillInfo.address),
+        identifier = identifier)
+
+    updateSpec
+        .given(searchQueryModel.patientPrefillInfoUpdated(patientPrefillInfo).additionalIdentifierUpdated(identifier))
+        .whenEvent(RegisterNewPatientClicked)
+        .then(assertThatNext(
+            hasNoModel(),
+            hasEffects(SaveNewOngoingPatientEntry(ongoingNewPatientEntry))
+        ))
+  }
+
+  @Test
+  fun `when register new patient is clicked and patient prefill info is not empty and dob is in wrong format, then save it in ongoing patient entry with dob in correct format`(){
+    val facility = TestData.facility()
+    val searchQueryModel = defaultModel
+        .facilityLoaded(facility)
+        .searchQueryChanged("Pat")
+
+    val dateOfBirthWithWrongFormat = LocalDate.parse("2/3/2012", DateTimeFormatter.ofPattern("d/M/yyyy"))
```

Comment:
I am not understanding how this is in the wrong format? If we expect the date in the NHID payload to follow a certain pattern, then yes, that can be wrong.

However, here we are creating a valid `LocalDate` and it can never be in the wrong format at all.

Reply:
Here, what you are actually testing is whether `LocalDate` is using the `DateTimeFormatter` passed to the constructor which we should not be testing since we don't own it.

Reply:
This is because we are saving the `dateOfBirth` that is being returned in the json response as `LocalDate` type. But, the format we want to prefill the `dateOfBirth` in `PatientEntryScreen` is in `dd/MM/yyyy` which is of `String` type. Which is what I'm trying to test here. If, the `LocalDate` is being converted into the correct format or not.

Reply:
Yeah, but this is the wrong place to do it. The place this needs to be tested is at the time of scanning the QR code, because once you parse a `LocalDate`, it is the same regardless of what format the string originally was,

Reply:
As mentioned earlier, the only thing this test is doing is verifying the behaviour of `LocalDate.parse`, which is something we should not be testing.